src/NetSparkle/AppCastHandlers/XMLAppCast.cs

```diff
@@ -182,66 +187,110 @@ protected virtual void ParseAppCast(string appCast)
         }
 
         /// <summary>
-        /// Returns sorted list of updates between current installed version and latest version in <see cref="Items"/>. 
-        /// Currently installed version is NOT included in the output.
+        /// Check if an AppCastItem update is valid, according to platform, signature requirements and current installed version number.
+        /// In the case where your app implements a downgrade strategy, e.g. when switching from a beta to a 
+        /// stable channel - there has to be a way to tell the update mechanism that you wish to ignore 
+        /// the beta AppCastItem elements, and that the latest stable element should be installed.  
         /// </summary>
-        /// <returns>A list of <seealso cref="AppCastItem"/> updates that could be installed</returns>
-        public virtual List<AppCastItem> GetAvailableUpdates()
+        /// <param name="installed">the currently installed Version</param>
+        /// <param name="signatureNeeded">whether or not a signature is required</param>
+        /// <param name="item">the AppCastItem under consideration, every AppCastItem found in the appcast.xml file is presented to this function once</param>
+        /// <returns>MatchingResult.MatchOk if the AppCastItem should be considered as a valid target for installation.</returns>
+        public MatchingResult IsMatchingUpdate(Version installed, bool signatureNeeded, AppCastItem item)
```

Comment:
Let's call this, `IsValidUpdate` instead of `IsMatchingUpdate`.

Reply:
I'm not 100% convinced.  The updates are all valid, otherwise they wouldn't be in the appcast.  I viewed this as a matter of whether the current install context matches the update... e.g. am I processing a beta version of the entire appcast file and I want to discard some of them because I'm trying to install the latest stable?  Then the idea of a match fits.  

Let me know if you still want to rename it.